src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -223,6 +225,7 @@ private void abort(RepairSegment segment, JmxProxy jmxConnection) {
   @Override
   public void handle(int repairNumber, ActiveRepairService.Status status, String message) {
     synchronized (condition) {
+      Thread.currentThread().setName(clusterName);
```

Comment:
JmxProxy.handleNotification is the only place where RepairStatusHandler.handle is ever called, outside a couple of tests (which I could change if necessary). The JMX thread calls JmxProxy.handleNotification, JmxProxy being a NotificationListener. It seems very redundant to me when you set the thread name again and adding clusterName to SegmentRunner (and RepairStatusHandler) just for that.